packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -547,8 +547,8 @@ function computeShadowMode(
     return shadowMode;
 }
 
-function assertIsVM(obj: any): asserts obj is VM {
-    if (isNull(obj) || !isObject(obj) || !('renderRoot' in obj)) {
+function assertIsVM(obj: unknown): asserts obj is VM {
+    if (!isObject(obj) || isNull(obj) || !('renderRoot' in obj)) {
```

Comment:
With the updated predicate, the original ordering excluded null and then added it straight back in, causing the third check to complain about null. Updating the order helps the type system correctly infer what's going on.